packages/itwin/grouping-mapping-widget/src/components/SharedComponents/ActionPanel.tsx

```diff
@@ -24,7 +25,7 @@ const ActionPanel = ({ onSave, onCancel, isCancelDisabled = false, isSavingDisab
           Save
         </Button>
         {onCancel && (
-          <Button styleType="default" type="button" id="cancel" onClick={onCancel} disabled={isCancelDisabled || isLoading}>
+          <Button styleType="default" type="button" id="cancel" onClick={onCancel} disabled={isCancelDisabled || isLoading} onClickCapture={onCancelCapture}>
```

Comment:
On second thought, this might not be enough. When you turn on the Calculated Properties visualization, it does not clear out when you save or toggle it off.

Reply:
Added onSaveCapture to clear the visualization on save, I've been messing around with visualization logic in CalculatedPropertyActionWithVisualst.tsx and it's not as simple as I though, changing the current implementation affects the behavior of the visualization, it seems like this needs to be a PR in its own, specially because `undefined` is now an option. 
Some thoughts on this: 
- I think the logic could be refactor a little bit, I think two of the useEffects that have the same falsely conditional could be merged together. 
- What would we want the visuals to be when `No Calculated Property` is selected?
The reason why it can't be in a useEffect now is because it would override the color group toggle (if on) from the property menu and clear the visuals, `calcualtedPropertyType` is `undefined` by default before any selection.
- If `No Calculated Property` is selected, would we want to clear out the group coloring all together or just the Spatial Data? If clearing it all, then it would override the previous toggle (if on) like mentioned. 
- What would be behavior be if the toggleSwitch is On but the expandable block is closed? 
- It's like the `undefined` selection and toggleSwitch are competing for control of the visualization logic so it makes it a bit tricky to reset the visuals, but the main concern is how it should behave.

Reply:
Oh I see, why it's confusing. So I double checked the current behavior and it is incorrect (before this PR). I must changed the behavior accidently when I switched to react-query. I have an older build that I pulled up before this.
The behavior was like this:

1. The PropertyMenu should remember the GroupToggle setting. If we leave the PropertyMenu, it should clear out the Color Group.
2. Going into the Calculated Property Action would clear out the visuals.
3. But leaving Calculated Property Action and going back to the Property Menu would rehilite the group if toggled. 

Reason for this was because we only wanted to hilite one element to show its Spatial Data. Even if ColorGroup is toggled (on). 

I think the behavior now should be.

- When leaving the PropertyMenu it should clear out its Group hilite (when it unmounts). But remember the toggle (should already be implemented). 
- Visualize Calculated Properties should also clear out everything when toggled on and off, or when unmounting the component (cleaning itself up and going away either through cancel or save or some other way). Selecting `No Calculated Property` can either clear out everything or the Spatial Data, up to you, I'm fine either way. 
- Now when the the PropertyMenu comes back it should remember the toggle selection and rehilite right away or not.

Reply:
Behavior of visuals accomplished. 
- Leaving property menu clears our visuals but it remembers the toggle if on when back to menu. 
- Visuals appear with toggle on and clear out when off, visuals also clear out when block closes(but remembers the selection) and when `No calculated Property` is selected, if  `No Calculated Property` is selected, it will close the block and clear visuals, visual will also clear out on onSave and onCancel.